primitiveMeshGen.py

```diff
@@ -17,6 +17,56 @@ def pol2cart(rad,theta):
   y = rad*np.sin(theta)
   return x,y
 
+#generate shed object
+def shedGen(height, length, theta, width):
+    xinterval = np.cos(theta) * width / 2
+    yinterval = length / 2
+    roof_height = xinterval * np.tan(theta)
+    vertices = []
+    vertices.append([-xinterval, yinterval, roof_height])
+    vertices.append([-xinterval, -yinterval, roof_height])
+    vertices.append([xinterval, -yinterval, -roof_height])
+    vertices.append([xinterval, yinterval, -roof_height])
+
+    vertices.append([-xinterval, yinterval, -height])
+    vertices.append([-xinterval, -yinterval, -height])
+    vertices.append([xinterval, -yinterval, -height])
+    vertices.append([xinterval, yinterval, -height])
+
+    faces = [[0, 1, 2, 3],
+             [4, 7, 6, 5],
+             [0, 4, 5, 1],
+             [1, 5, 6, 2],
+             [2, 6, 7, 3],
+             [3, 7, 4, 0]]
+    return vertices, faces
+
+#generate gable object
```

Comment:
Again, some documentation would help make this more clear. 

This one seems to put the peak of the roof at the origin.  I think it would be better to put either the base of the roof or the base of the building at the origin and be consistent across all roof types.  However, this just a convention and is up for debate.